File: csharp/acronym/Acronym.cs
using System;
using System.Linq;
using System.Text;

public static class Acronym
{
    public static string Abbreviate(string phrase)
    {
        phrase = phrase.ToUpper();
        var sb = new StringBuilder();

        foreach (var s in phrase.Split(new char[] {' ', '-', '_'}))
        {
            if (s.Length == 0) { continue; }

            char c = s[0];
            
            if (Char.IsLetter(c))
            {
                sb.Append(c);
            }
        }

        return sb.ToString();
    }
}

File: java/acronym/src/main/java/Acronym.java
class Acronym {

    String acronym;

    Acronym(String phrase) {
        acronym = findAcronym(phrase);
    }

    String get() {
        return acronym;
    }

    private String findAcronym(String phrase) {

        String[] words = phrase.split(" |-|_");
        var sb = new StringBuilder();

        for (var word : words) {

            if (word.isEmpty()) { continue; }
            sb.append(word.charAt(0));
        }

        return sb.toString().toUpperCase();
    }
}
